packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/util.rs

```diff
@@ -29,13 +29,26 @@ pub fn create_data_query_if_match_extend_source(
 
 /// Convert string to boolean
 ///
-/// The word "true" (case-insensitive) or an empty string become true.
-/// All other strings become false.
-///
-/// TODO: the empty string is try so that attribute="" sets the boolean attribute to true.
-/// Are there other cases where we'd want the empty string to be false?
-/// For example, an empty `<textInput>` becomes true with this algorithm,
-/// which may be confusing for authors. Not sure how to treat that case differently.
-pub fn string_to_boolean(s: &str) -> bool {
-    s.eq_ignore_ascii_case("true") || s.is_empty()
+/// The word "true" (case-insensitive) becomes `true`.
+/// If `empty_is_true` is `true`, then an empty string also becomes `true`.
+/// All other strings become `false`.
+pub fn string_to_boolean(s: &str, empty_is_true: bool) -> bool {
```

Comment:
Haven't seen the use case yet, but I don't like the `empty_is_true` parameter.

Reply:
For an attribute, like `<textInput disabled>`, we want the empty string that ends up getting created for `disabled`  to become `true`. For `<boolean>$t</boolean>`, if `$t` resolves to an empty string, we want the result to be false. So, I made a flag to determine if we are in an attribute, which changes the behavior of `string_to_boolean`.

The question is: do we want `<textInput disabled="" />` and `<textInput disabled="$t">`, where `$t` resolves to an empty string to make disabled be `true`? It is currently `false` in version 0.6. I was going to allow it to be `true` given our approach for handling empty attributes. I guess I could shop that around. If we want to maintain the behavior of version 0.6, then we need a different solution which I think will involve parsing empty attribute differently.

Reply:
I suggest a different function if you want the other behavior. But, really, you have a `attr_to_boolean` function, which treats the empty attr differently than other cases.

Reply:
have separate functions now